client/engine/engine.go

```diff
@@ -721,3 +727,22 @@ func (e *Engine) recordMessageMetrics(message protocols.Message) {
 	e.metrics.RecordQueueLength(fmt.Sprintf("msg_payload_size,sender=%s,receiver=%s", e.store.GetAddress(), message.To), totalPayloadsSize)
 	e.metrics.RecordQueueLength(fmt.Sprintf("msg_size,sender=%s,receiver=%s", e.store.GetAddress(), message.To), len(raw))
 }
+
+func (e *Engine) checkError(err error) {
+	if err != nil {
+
+		e.logger.Err(err).Msgf("%s, error in run loop", e.store.GetAddress())
+
+		for _, nonFatalError := range nonFatalErrors {
+			if errors.Is(err, nonFatalError) {
+				return
+			}
+		}
+
+		<-time.After(1000 * time.Millisecond) // We wait for a bit so the previous log line has time to complete
+		// TODO do not panic if in production.
```

Comment:
We could remove this TODO now, I think -- point being, we have pinched off errors that actually do require a panic in production.